common/usb_cdc.c

```diff
@@ -44,20 +44,27 @@
 #define AT91C_EP_OUT_SIZE 0x40
 #define AT91C_EP_IN_SIZE  0x40
 
+// Language must always be 0.
+#define STR_LANGUAGE_CODES 0x00
+#define STR_MANUFACTURER   0x01
+#define STR_PRODUCT        0x02
+
 static const char devDescriptor[] = {
 	/* Device descriptor */
 	0x12,      // bLength
 	0x01,      // bDescriptorType
 	0x00,0x02, // Complies with USB Spec. Release (0200h = release 2.0)
 	0x02,      // bDeviceClass:    (Communication Device Class)
-	0x00,      // bDeviceSubclass: (unused at this time)
-	0x00,      // bDeviceProtocol: (unused at this time)
+	0x02,      // bDeviceSubclass: (Abstract Control Model)
+	           // Allows use on Win10+ without INF file. Reference:
+	           // https://msdn.microsoft.com/en-us/library/windows/hardware/dn707976%28v=vs.85%29.aspx
+	0xff,      // bDeviceProtocol: (Vendor-specific interface)

```

Comment:
Similar. This would be a setting for bInterfaceProtocol (which you changed below from 0x01 to 0x00).

Reply:
Changing the InterfaceProtocol (not DeviceProtocol) to 0xff causes Linux to treat it as a RNDIS Ethernet adaptor:

- https://github.com/torvalds/linux/blob/master/drivers/usb/class/cdc-acm.c#L1907
- http://elixir.free-electrons.com/linux/latest/source/drivers/net/usb/rndis_host.c#L625

So while it seems to be technically correct, Microsoft screws us up here.